server/index.js

```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
       res.send(err);
     } else {
       res.send(docs);
-    }    
+    }
   });
 });
 
-app.get('/api/bpowertest', (req, res) =>{
-  dpower.find((err, docs) => {
+app.get('/api/buytest/one/:id', (req, res) => {
+  const id = JSON.parse(req.params.id);
+  modalBuy.read(id, (err, doc) => {
     if (err) {
-      res.send(err);
+      res.status(500).send(err);
```

Comment:
This is probably ok error handling, but you might also try looking into express's built-in error handling to see what is standard

Reply:
Great. Taking a look.